matrix_registration/api.py

```diff
@@ -221,6 +223,7 @@ def register():
     else:
         server_name = config.config.server_name
         pw_length = config.config.password['min_length']
+        uname_regex = config.config.username['validation_regex'] or ''
```

Comment:
Btw: this was meant for backwarts compat but it should throw a error after trying to index username if it doesn't exist; or does python doesn't have nullpoiner exceptions?

Reply:
well it would probably just throw a `AttributeError`. I don't think it will ever get to `or ''`.
So far my policy has been that this is a <1.0.0 so nothing is particularly stable and new options get added and need to be manually added to the config file